RHIDP-2703: update oidc configuration documentation (#293)
redhat-developer/red-hat-developers-documentation-rhdh: modules/getting-started/proc-configuring-keycloak-with-oidc.adoc

      #tokenEndpointAuthMethod: ${KEYCLOAK_TOKEN_ENDPOINT_METHOD}
      #tokenSignedResponseAlg: ${KEYCLOAK_SIGNED_RESPONSE_ALG}
      #scope: ${KEYCLOAK_SCOPE}

>       ## It is recommended to use the `preferredUsernameMatchingUserEntityName` resolver if using the keycloak-backend plugin

Reviewer: @Zaperex Can you please tell me why this is recommended or what will happen if the user does not use this resolver so that I can update this part? 

We strive to avoid saying the word "recommend" in RH docs; instead, we tell the user to perform the recommended action and state WHY they should (or should not) do something.

Author: So the original change was made to fix [RHIDP-2272](https://issues.redhat.com/browse/RHIDP-2272) where the OIDC auth provider did not resolve properly when ingesting users/groups using the keycloak-backend plugin using the default keycloak configurations. So if the user does not use the `preferredUsernameMatchingUserEntityName` there's a good chance they will not be able to login with OIDC. It all depends on how their OIDC provider is setup though, but this is made with the assumption that is is connected to keycloak.

Reviewer: ```suggestion
      ## If you are using the keycloak-backend plugin, use the `preferredUsernameMatchingUserEntityName` resolver to avoid a login error.
```
this is probably sufficient
